<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="6" ignored="0" total="8" passed="2">
  <reporter-output>
  </reporter-output>
  <suite name="Erail Application Testing" duration-ms="13365" started-at="2020-01-20T12:29:59Z" finished-at="2020-01-20T12:30:12Z">
    <groups>
    </groups>
    <test name="ErailHomePageVerification" duration-ms="13365" started-at="2020-01-20T12:29:59Z" finished-at="2020-01-20T12:30:12Z">
      <class name="com.erail.qa.Tests.HomePageTests">
        <test-method status="PASS" signature="SetUpBeforeTest(java.lang.String)[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeTest" is-config="true" duration-ms="8818" started-at="2020-01-20T12:29:59Z" finished-at="2020-01-20T12:30:08Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeTest -->
        <test-method status="PASS" signature="SetUp()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUp" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUp -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="PASS" signature="VerifyHomePage()[pri:1, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="VerifyHomePage" duration-ms="128" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- VerifyHomePage -->
        <test-method status="PASS" signature="VerifyHomePage()[pri:1, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="VerifyHomePage" duration-ms="142" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- VerifyHomePage -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:08Z" finished-at="2020-01-20T12:30:08Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="FAIL" signature="SearchTrain(java.lang.String, java.lang.String)[pri:2, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SearchTrain" duration-ms="449" started-at="2020-01-20T12:30:08Z" data-provider="GetTestData" finished-at="2020-01-20T12:30:09Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[Mumbai]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[New Delhi]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Mumbai] but found [MumbaiMumbai]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Mumbai] but found [MumbaiMumbai]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:453)
at org.testng.Assert.assertEquals(Assert.java:463)
at com.erail.qa.Pages.HomePage.EnterFromToStation(HomePage.java:73)
at com.erail.qa.Tests.HomePageTests.SearchTrain(HomePageTests.java:62)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SearchTrain -->
        <test-method status="FAIL" signature="SearchTrain(java.lang.String, java.lang.String)[pri:2, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SearchTrain" duration-ms="449" started-at="2020-01-20T12:30:08Z" data-provider="GetTestData" finished-at="2020-01-20T12:30:09Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[Mumbai]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[New Delhi]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Mumbai] but found [MumbaiMumbai]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Mumbai] but found [MumbaiMumbai]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:453)
at org.testng.Assert.assertEquals(Assert.java:463)
at com.erail.qa.Pages.HomePage.EnterFromToStation(HomePage.java:73)
at com.erail.qa.Tests.HomePageTests.SearchTrain(HomePageTests.java:62)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SearchTrain -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="FAIL" signature="SearchTrain(java.lang.String, java.lang.String)[pri:2, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SearchTrain" duration-ms="507" started-at="2020-01-20T12:30:09Z" data-provider="GetTestData" finished-at="2020-01-20T12:30:09Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[New Delhi]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Karnataka]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [New Delhi] but found [New DelhiNew]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [New Delhi] but found [New DelhiNew]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:453)
at org.testng.Assert.assertEquals(Assert.java:463)
at com.erail.qa.Pages.HomePage.EnterFromToStation(HomePage.java:73)
at com.erail.qa.Tests.HomePageTests.SearchTrain(HomePageTests.java:62)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SearchTrain -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="FAIL" signature="SearchTrain(java.lang.String, java.lang.String)[pri:2, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SearchTrain" duration-ms="518" started-at="2020-01-20T12:30:09Z" data-provider="GetTestData" finished-at="2020-01-20T12:30:09Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[New Delhi]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Karnataka]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [New Delhi] but found [New DelhiNew]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [New Delhi] but found [New DelhiNew]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:453)
at org.testng.Assert.assertEquals(Assert.java:463)
at com.erail.qa.Pages.HomePage.EnterFromToStation(HomePage.java:73)
at com.erail.qa.Tests.HomePageTests.SearchTrain(HomePageTests.java:62)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SearchTrain -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="FAIL" signature="BestMacthTrain()[pri:3, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="BestMacthTrain" duration-ms="999" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:10Z">
          <exception class="org.testng.internal.thread.ThreadTimeoutException">
            <message>
              <![CDATA[Method com.erail.qa.Tests.HomePageTests.BestMacthTrain() didn't finish within the time-out 1000]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.internal.thread.ThreadTimeoutException: Method com.erail.qa.Tests.HomePageTests.BestMacthTrain() didn't finish within the time-out 1000
at java.net.SocketInputStream.socketRead0(Native Method)
at java.net.SocketInputStream.socketRead(Unknown Source)
at java.net.SocketInputStream.read(Unknown Source)
at java.net.SocketInputStream.read(Unknown Source)
at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137)
at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:153)
at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:282)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:165)
at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
at org.openqa.selenium.remote.internal.ApacheHttpClient.fallBackExecute(ApacheHttpClient.java:139)
at org.openqa.selenium.remote.internal.ApacheHttpClient.execute(ApacheHttpClient.java:87)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:161)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:646)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:416)
at org.openqa.selenium.remote.RemoteWebDriver.findElementByXPath(RemoteWebDriver.java:518)
at org.openqa.selenium.By$ByXPath.findElement(By.java:361)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:408)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at com.sun.proxy.$Proxy15.isDisplayed(Unknown Source)
at org.openqa.selenium.support.ui.ExpectedConditions.elementIfVisible(ExpectedConditions.java:302)
at org.openqa.selenium.support.ui.ExpectedConditions.access$100(ExpectedConditions.java:44)
at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:288)
at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:285)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:209)
at com.erail.qa.Pages.HomePage.GetBestMatchTrains(HomePage.java:107)
at com.erail.qa.Tests.HomePageTests.BestMacthTrain(HomePageTests.java:69)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.internal.thread.ThreadTimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- BestMacthTrain -->
        <test-method status="PASS" signature="SetUpBeforeMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="SetUpBeforeMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:09Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUpBeforeMethod -->
        <test-method status="FAIL" signature="BestMacthTrain()[pri:3, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="BestMacthTrain" duration-ms="999" started-at="2020-01-20T12:30:09Z" finished-at="2020-01-20T12:30:10Z">
          <exception class="org.testng.internal.thread.ThreadTimeoutException">
            <message>
              <![CDATA[Method com.erail.qa.Tests.HomePageTests.BestMacthTrain() didn't finish within the time-out 1000]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.internal.thread.ThreadTimeoutException: Method com.erail.qa.Tests.HomePageTests.BestMacthTrain() didn't finish within the time-out 1000
at java.net.SocketInputStream.socketRead0(Native Method)
at java.net.SocketInputStream.socketRead(Unknown Source)
at java.net.SocketInputStream.read(Unknown Source)
at java.net.SocketInputStream.read(Unknown Source)
at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137)
at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:153)
at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:282)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:165)
at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
at org.openqa.selenium.remote.internal.ApacheHttpClient.fallBackExecute(ApacheHttpClient.java:139)
at org.openqa.selenium.remote.internal.ApacheHttpClient.execute(ApacheHttpClient.java:87)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:161)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:646)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:416)
at org.openqa.selenium.remote.RemoteWebDriver.findElementByXPath(RemoteWebDriver.java:518)
at org.openqa.selenium.By$ByXPath.findElement(By.java:361)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:408)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at com.sun.proxy.$Proxy15.isDisplayed(Unknown Source)
at org.openqa.selenium.support.ui.ExpectedConditions.elementIfVisible(ExpectedConditions.java:302)
at org.openqa.selenium.support.ui.ExpectedConditions.access$100(ExpectedConditions.java:44)
at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:288)
at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:285)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:209)
at com.erail.qa.Pages.HomePage.GetBestMatchTrains(HomePage.java:107)
at com.erail.qa.Tests.HomePageTests.BestMacthTrain(HomePageTests.java:69)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.internal.thread.ThreadTimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- BestMacthTrain -->
        <test-method status="PASS" signature="TearDown()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDown" is-config="true" duration-ms="668" started-at="2020-01-20T12:30:10Z" finished-at="2020-01-20T12:30:11Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDown -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="1" started-at="2020-01-20T12:30:10Z" finished-at="2020-01-20T12:30:10Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="TearDownAfterMethod()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterMethod" is-config="true" duration-ms="0" started-at="2020-01-20T12:30:10Z" finished-at="2020-01-20T12:30:10Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterMethod -->
        <test-method status="PASS" signature="TearDownAfterTest()[pri:0, instance:com.erail.qa.Tests.HomePageTests@52f759d7]" name="TearDownAfterTest" is-config="true" duration-ms="1685" started-at="2020-01-20T12:30:11Z" finished-at="2020-01-20T12:30:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDownAfterTest -->
      </class> <!-- com.erail.qa.Tests.HomePageTests -->
    </test> <!-- ErailHomePageVerification -->
  </suite> <!-- Erail Application Testing -->
</testng-results>
